### PR TITLE
Added Bandit security analysis.

### PR DESCRIPTION
**Pull request by garyray-k**
Included in poetry install. new circle run as well.

# Description of changes
This adds the Bandit tool to the project, which scans for known security issues laid out by the tool. The tool is open source and can be found [here](https://github.com/PyCQA/bandit). It was found on the [OWASP](https://www.owasp.org/index.php/Source_Code_Analysis_Tools) website for Source Code Analysis Tools. 
There was a baseline Bandit report ran with json output file created and date-stamped for future reference. This baseline was created because currently Bandit has several false positives for code optimization issues on "asserts" in the tests. It also flags a hardcoded temp or tmp directory, again in the tests. Since this is just in tests, I chose to baseline all of these issues so future comparisons will ignore them and only new "issues" will arise.

# Issue Resolved
OC Projects Security initiative. Bandit is one step of many still in development.

## Screenshots/GIFs
N/A

### CODE REVIEW

File: pyproject.toml [hunk 1]
```diff
@@ -25,6 +25,7 @@ mailchimp3 = "^3.0"
 python-dotenv = "^0.10.1"
 requests = "^2.21"
 sentry-sdk = "^0.7.14"
+bandit = "~1.5.1"
```
Comment (AllenAnthes) [inline pyproject.toml]: We only need this declared as a dev-dependency since it's not needed at runtime.

File: pyproject.toml [hunk 2]
```diff
@@ -38,6 +39,7 @@ pytest-mock = "^1.10"
 pytest-pythonpath = "^0.7.3"
 flake8 = "^3.7"
 flake8-bugbear = "^19.3"
+bandit = "~1.5.1"
```
Comment (AllenAnthes) [inline pyproject.toml]: If you could move this entry up to keep things in alphabetical order, for no other reason than to appease my OCD, I'd really appreciate it  😊

**Review by AllenAnthes — CHANGES_REQUESTED** [review]

**Review by garyray-k — COMMENTED** [review]
fat fingered the commit message but the changes were made.

File: .circleci/config.yml [hunk 3]
```diff
@@ -94,5 +94,10 @@ jobs:
             poetry run coverage xml
             /usr/local/bin/cc-test-reporter after-build -t coverage.py
 
+      - run:
+        name: Run Bandit security analysis
+          command: |
```
Comment (AllenAnthes) [inline .circleci/config.yml:100]: Looks like an indentation error failed the build here.  You can use their CLI tool for validating the config locally before committing.

https://circleci.com/docs/2.0/local-cli/#validate-a-circleci-config

**Review by AllenAnthes — CHANGES_REQUESTED** [review]

**Comment by garyray-k** [comment]
Looks like the same issue I was getting when I tried using CircleCI on my forked repo.
As far as I can tell, it runs fine until it hits the `/usr/local/bin/cc-test-reporter after-build -t coverage.py` line and then spits out the following. (this is just after the `poetry run coverage report` terminal output)
```
Error: you must supply a CC_TEST_REPORTER_ID ENV variable or pass it via the -r flag
Usage:
  cc-test-reporter after-build [flags]

Flags:
  -s, --batch-size int               batch size for source files (default 500)
  -e, --coverage-endpoint string     endpoint to upload coverage information to (default "https://api.codeclimate.com/v1/test_reports")
  -t, --coverage-input-type string   type of input source to use [clover, cobertura, coverage.py, excoveralls, gcov, gocov, jacoco, lcov, simplecov]
      --exit-code int                exit code of the test run
  -r, --id string                    reporter identifier
      --insecure                     send coverage insecurely (without HTTPS)
  -p, --prefix string                the root directory where the coverage analysis was performed (default "/home/circleci/project")

Global Flags:
  -d, --debug   run in debug mode

Exited with code 255
```

**Comment by garyray-k** [comment]
Screen Shot 2019-05-23 at 6 49 45 AM

**Review by AllenAnthes — APPROVED** [review]

**Pull request merged**